src/wasm.h

```diff
@@ -1233,13 +1234,104 @@ std::ostream& Expression::print(std::ostream &o, unsigned indent) {
   return o;
 }
 
+template<typename SubType, typename ReturnType>
```

Comment:
I would do `typename ReturnType = void` here, since it seems to be the default when you use it.